File: CS50/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
#include <stdbool.h>

int main(void)
{
    string texto = get_string("Text: ");

    int letras = 0;
    int frase = 0;

    int palavras = 0;
    bool em_palavra = false;

    for (int i = 0, n = strlen(texto); i < n; i++)
    {
        if (isalpha(texto[i])) letras++;
        if (texto[i] == '.' || texto[i] == '!' || texto[i] == '?') frase++;

        if (isspace(texto[i]))
        {
            em_palavra = false;
        }
        else if (!em_palavra)
        {
            palavras++;
            em_palavra = true;
        }
    }

    float L = (float) letras / palavras * 100;
    float S = (float) frase / palavras * 100;

    int Grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (Grade < 1) printf("Before Grade 1\n");
    else if (Grade >= 16) printf("Grade 16+\n");
    else printf("Grade %i\n", Grade);
}

File: pset6/readability.py
from cs50 import get_string

texto = get_string("Qual o texto: ")

palavras = 0
letras= 0
frases = 0

for ch in texto:
    if ch.isalpha(): letras +=1
    elif ch.isspace(): palavras +=1
    elif ch in ".!?'": frases += 1

L = letras / palavras * 100
S = frases / palavras * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1 ")

elif index > 16:
    print("Grade 16+")

else:
  print("Grade: " , index)




